pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -837,6 +862,47 @@ func (c *Controller) updateEDS(ep *v1.Endpoints, event model.Event) {
 	_ = c.XDSUpdater.EDSUpdate(c.ClusterID, string(hostname), ep.Namespace, endpoints)
 }
 
+// called when the overall mTLS ready status of endpoints have changed for a service and the cached Service object must be updated
+func (c *Controller) updateSvcMtlsReady(svc *model.Service) {
+	// TODO gihanson does the mTLS ready state impact external name services?
```

Comment:
it absolutely does. Especially when I back the external name service by k8s services.